elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    DOWN: 40,
+    ESCAPE: 27,
+    TAB: 9,
+    UP: 38,
+  };
+
+  class HowtoMenu extends HTMLElement {
+    /**
+     * A getter for the first child which is a menuitem.
+     */
+    get _firstMenuItem() {
+      let child = this.firstElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * A getter for the last child which is a menuitem.
+     */
+    get _lastMenuItem() {
+      let child = this.lastElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Checks if a node is a 'menuitem', 'menuitemcheckbox' or 'menuitemradio'.
+     */
+    _isMenuItem(node) {
+      let ariaRoles = ['menuitem', 'menuitemcheckbox', 'menuitemradio'];
+      return ariaRoles.indexOf(node.getAttribute('role') > -1);
+    }
+
+    /**
+     * Returns a menu item coming after the one passed as an argument, or null.
+     */
+    _nextMenuItem(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (this._isMenuItem(node)) {
+          return next;
+        }
+        next = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns a menu item coming before the one passed as an argument, or null.
+     */
+    _previousMenuItem(node) {
+      let prev = node.previousElementSibling;
+      while (prev) {
+        if (this._isMenuItem(node)) {
+          return prev;
+        }
+        prev = prev.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Handles keyboard interaction.
+     */
+    _onKeyDown(event) {
```

Comment:
Yes, in this version menuitems are not activateable (is this even a word?). Implementing submenus is a bigger deal so I'd rather do it once basic version lands.